src/cmds/qsub.c

```diff
@@ -2283,7 +1879,7 @@ interactive(void)
 		exit_qsub(1);
 	}
 	LeaveCriticalSection(&continuethread_cs);
-	strncpy(remote_ip, inet_ntoa(from.sin_addr), INET_ADDR_STRLEN);
+	strncpy(remote_ip, inet_ntoa(from.sin_addr), INET_ADDR_STRLEN); /* safe because remote_ip was initialized with zeroes */
```

Comment:
Since `sizeof(remote_ip) = INET_ADDR_STRLEN + 1` and here `strncpy()` will copy at most `INET_ADDR_STRLEN` bytes, wouldn't that guarantee that the last byte of `remote_ip` won't be overwritten? So if it was initialized with zeroes, then the last byte is guaranteed to be zero. Regardless, I agree that this line is confusing, so I'll just change it to snprintf with `sizeof(remote_ip)` as the size parameter.